src/app/test/test.component.html

```diff
@@ -3,25 +3,37 @@
-      <app-button icon="fa fa-cog" title="Options" (click)="openSettingsModal()"></app-button>
-      <app-button data-cy-test="selectAll" title="Select All" text="Select All"
-                  (click)="checkAll()"></app-button>
-      <app-button data-cy-test="deselectAll" title="Deselect All" text="Deselect All"
-                  (click)="uncheckAll()"></app-button>
-      <app-button data-cy-test="deleteSelected" icon="fa fa-trash" title="Delete Selected"
-                  (click)="openDeleteModal()"></app-button>
-      <app-button text="Clone" title="Clone report" (click)="openCloneModal()"></app-button>
-      <app-button data-cy-test="copySelected" icon="fa fa-copy" title="Copy Selected"
-                  (click)="copySelected()"></app-button>
-      <app-button data-cy-test="downloadBinary" icon="fa fa-download" title="Download Binary"
-                  (click)="downloadSelected()"></app-button>
-      <button data-cy-test="upload" type="button" class="btn btn-info my-2 mx-1 px-3"
-              title="Upload" (click)="uploadFileTest.click()">
+    <div class="buttons pt-1">
+      <button data-cy-test="runAll" class="btn btn-info" title="Run All" (click)="runSelected()"><i
+        class="fa fa-play"></i></button>
+      <button data-cy-test="refresh" class="btn btn-info" title="ReloadTestReports"
+              (click)="loadData('')"><i class="fa fa-refresh"></i></button>
+      <button class="btn btn-info" title="Reset" (click)="resetRunner()">Reset</button>
+      <button class="btn btn-info" title="Options" (click)="openSettingsModal()"><i class="fa fa-cog"></i></button>
+      <button
+        data-cy-test="selectAll"
+        class="btn btn-info"
+        title="Select All"
+        (click)="setCheckedForAllReports(true)"
+      >Select All
+      </button>
+      <button data-cy-test="deselectAll"
+              class="btn btn-info"
+              title="Deselect All"
+              (click)="setCheckedForAllReports(false)">
+        Deselect All
+      </button>
+      <button data-cy-test="deleteSelected" class="btn btn-info" title="Delete Selected"
+              (click)="openDeleteModal()"><i class="fa fa-trash"></i></button>
+      <button class="btn btn-info" title="Clone report"
+              (click)="openCloneModal()">Clone
+      </button>
+      <button data-cy-test="copySelected" class="btn btn-info" title="Copy Selected"
+              (click)="copySelected()"><i class="fa fa-copy"></i></button>
+      <button data-cy-test="downloadBinary" class="btn btn-info" title="Download Binary"
+              (click)="downloadSelected()"><i class="fa fa-download"></i></button>
+      <button data-cy-test="upload" type="button" class="btn btn-info" title="Upload"
+              (click)="uploadFileTest.click()">
         <i class="fa fa-upload"></i>
         <input #uploadFileTest data-cy-test="uploadFile" type="file" id="uploadFileTest"
                (change)="uploadReport($event)">
```

Comment:
Is it necessary to have this extra input element? Wouldn't it be better to just the button or just the input element or am I missing something?

Reply:
Yeah the input element is there to allow native browser file uploads

Reply:
Alright. Good to know.